### PR TITLE
feat: Add Pipedrive ListMetadata connector

### PR DESCRIPTION
**Pull request by Jkarage**
This adds the PipeDrive ListMetadata connector:

- Attempts getting the metadata through the API, If it fails (mostly due to 0 records returned), it will use the static schemas.json file.

Test Outputs:
Screenshot 2024-10-14 at 16 23 17

### CODE REVIEW

**Review by Cobalt0s — COMMENTED** [review]
Looks great!

Metadata with schema file fallback is great.

Same thing as with the other PR, would be nice to have mock tests. 
One would immitate API response with response check and the other would have API call fail so the connector would fallback to static schemas.

**Comment by laurenzlong** [comment]
@Cobalt0s  please re-review when you have a chance.

File: providers/pipedrive/metadata_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		},
+		{
+			Name:  "Zero records returned from server",
+			Input: []string{"activities"},
+			Server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusOK)
+				_, _ = w.Write([]byte(zeroRecords))
+			})),
+			Expected: &common.ListObjectMetadataResult{
```
Comment (Cobalt0s) [inline providers/pipedrive/metadata_test.go:63]: You could shrink the output by comparing only the subset of fields: 
```
Comparator: func(baseURL string, actual, expected *common.ListObjectMetadataResult) bool {
    return mockutils.MetadataResultComparator.SubsetFields(actual, expected)
},
```

File: providers/pipedrive/objectNames.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package pipedrive
+
+import (
+	"github.com/amp-labs/connectors/common/handy"
+)
+
+// ObjectNameToResponseField maps ObjectName to the response field name which contains that object.
+var ObjectNameToResponseField = handy.NewDefaultMap(map[string]string{}, //nolint:gochecknoglobals
```
Comment (Cobalt0s) [inline providers/pipedrive/objectNames.go:8]: This declaration is not needed, for OpenAPI tool please use: `api3.IdenticalObjectCheck` which does exactly as the method.
Ex:
```
objects, err := explorer.GetBasicReadObjects(
	ignoreEndpoints, nil, nil, api3.IdenticalObjectCheck,
)
```

**Review by Cobalt0s — APPROVED** [review]
LGTM

File: providers/pipedrive/metadata_test.go (hunk 1 above)
Comment (Jkarage) [reply providers/pipedrive/metadata_test.go:63]: Cool, Looking into updating/cleaning these tests, As am working on adding Mock Tests for the other connectors.

File: scripts/oauth/token.go [hunk 3]
```diff
@@ -272,7 +272,7 @@ func setup() *OAuthApp {
 
 	substitutions, err := registry.GetMap(SubstitutionsFieldName)
 	if err != nil {
-		slog.Warn("no substitutions, ensure that the provider info doesn't have any {{variables}}", "error", err)
+		slog.Warn("no substitutions, ensure that the provider info doesn't have any variables", "error", err)
```
Comment (Cobalt0s) [inline scripts/oauth/token.go:275]: I understand why this line was changed as I had troubles myself. But the brackets have to stay as it is referes to catalog variables, ex: `{{.workspace}}`.

**Pull request merged**